src/main/java/io/supertokens/webserver/api/session/HandshakeAPI.java

```diff
@@ -49,35 +50,34 @@ public String getPath() {
 
     @Override
     protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        // API is tenant specific
         try {
             JsonObject result = new JsonObject();
             result.addProperty("status", "OK");
 
+            TenantIdentifier tenantIdentifier = this.getTenantIdentifierWithStorageFromRequest(req);
+
             result.addProperty("jwtSigningPublicKey",
                     new Utils.PubPriKey(
-                            AccessTokenSigningKey.getInstance(this.getTenantIdentifierWithStorageFromRequest(req).toAppIdentifier(),
+                            AccessTokenSigningKey.getInstance(tenantIdentifier.toAppIdentifier(),
                                     main).getLatestIssuedKey().value).publicKey);
             result.addProperty("jwtSigningPublicKeyExpiryTime",
-                    AccessTokenSigningKey.getInstance(this.getTenantIdentifierWithStorageFromRequest(req).toAppIdentifier(), main)
+                    AccessTokenSigningKey.getInstance(tenantIdentifier.toAppIdentifier(), main)
                             .getKeyExpiryTime());
 
             if (!super.getVersionFromRequest(req).equals("2.7") && !super.getVersionFromRequest(req).equals("2.8")) {
-                List<KeyInfo> keys = AccessTokenSigningKey.getInstance(this.getTenantIdentifierWithStorageFromRequest(req).toAppIdentifier(),
-                                main)
+                List<KeyInfo> keys = AccessTokenSigningKey.getInstance(tenantIdentifier.toAppIdentifier(), main)
                         .getAllKeys();
                 JsonArray jwtSigningPublicKeyListJSON = Utils.keyListToJson(keys);
                 result.add("jwtSigningPublicKeyList", jwtSigningPublicKeyListJSON);
             }
 
```

Comment:
are you sure all the props here are not per app?

Reply:
properties seems to be fine